lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul


# Test case 1: Basic test with 2x2 matrices
>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
array([[ 7, 10],
       [15, 22]])

# Test case 2: Different sized matrices
>>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
array([[13, 16]])

# Test case 3: Identity matrix
>>> lazy_matrix_mul(np.eye(3), [[1, 2, 3], [4, 5, 6], [7, 8, 9]])
array([[ 1.,  2.,  3.],
       [ 4.,  5.,  6.],
       [ 7.,  8.,  9.]])

# Test case 4: Zero matrix
>>> lazy_matrix_mul([[0, 0], [0, 0]], [[1, 2], [3, 4]])
array([[0, 0],
       [0, 0]])

# Test case 5: Error case - incompatible dimensions
>>> lazy_matrix_mul([[1, 2]], [[3, 4]])  # doctest: +IGNORE_EXCEPTION_DETAIL
Traceback (most recent call last):
    ...
ValueError: shapes (1,2) and (1,2) not aligned: 2 (dim 1) != 1 (dim 0)

